repos/system_upgrade/el7toel8/models/pcidevices.py

```diff
@@ -17,9 +57,73 @@ class PCIDevice(Model):
     driver = fields.Nullable(fields.String())
     modules = fields.Nullable(fields.List(fields.String()))
     numa_node = fields.Nullable(fields.String())
+    pci_id = fields.String()
 
 
 class PCIDevices(Model):
     topic = SystemInfoTopic
 
     devices = fields.List(fields.Model(PCIDevice))
+
+
+class RestrictedPCIDevice(Model):
+    """
+    Model to represent knwon restrictions of the given PCI devices.
+
+
+    pci_id - unsupported pci_ids. It has the following
+        structure: {Vendor}:{Device}:{SVendor}:{SDevice}, where all these 4
+        parameters are numeric ids (see shell command spci -vmmkn). If SVendor
+        and SDevice fields do not exist, then pci_id has the following structure:
+        {Vendor}:{Device}.
+    driver_name - the name of restricted driver
+    device_name - the name of restricted device
+    supported_{rhel_version} - 1 is supported on the given RHEL, 0 - not
+        supported
+    available_{rhel_version} - 1 is available on the given RHEL, 0 - not
+        available. it could be the driver is available, but not supported
+    comments - the field for comments
+    """
+    topic = SystemInfoTopic
+
+    pci_id = fields.String()
+    driver_name = fields.String()
+    device_name = fields.String()
+    available_rhel7 = fields.Integer()
+    supported_rhel7 = fields.Integer()
+    available_rhel8 = fields.Integer()
+    supported_rhel8 = fields.Integer()
+    available_rhel9 = fields.Integer()
+    supported_rhel9 = fields.Integer()
+    comment = fields.String()
+
+
+class RestrictedPCIDevices(Model):
+    """
+    Model to represent all known restricted PCI devices.
+
+    Restricted devices might be restricted based on either
+     - pci id (each particular device)
+     - driver name (all the family of the devices being served by the
+        particular driver).
+
+    However the data type, which represents how the pci id or driver name is
+    restricted is identical.
+
+    Thus both attributes has the same data type.
+
+    driver_names - is a container with the devices, which are restricted by
+        the driver name identifier
+    pci_ids - is a container with the devices, which are restricted by
+        the pci_id identifier
+    """
+    topic = SystemInfoTopic
+
+    driver_names = fields.List(
+        fields.Model(RestrictedPCIDevice),
+        default=(),
```

Comment:
The field should be list, not tuple.
```
>>> type(())
<class 'tuple'>
>>> type([])
<class 'list'>
```

```suggestion
        default=[],
```

Reply:
I will remove the default. Having the mutable object as a default is dangerous.

Reply:
@ZhukovGreen I didn't get this. so what it will be if nothing is set? I think it's not so valid argument when it is used like that on many other places.